pkg/framework/plugins/removeduplicates/removeduplicates.go

```diff
@@ -0,0 +1,299 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package removeduplicates
+
+import (
+	"context"
+	"fmt"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"math"
+	"reflect"
+	"sigs.k8s.io/descheduler/pkg/utils"
+	"sort"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/klog/v2"
+
+	"sigs.k8s.io/descheduler/pkg/apis/componentconfig"
+	"sigs.k8s.io/descheduler/pkg/descheduler/evictions"
+	podutil "sigs.k8s.io/descheduler/pkg/descheduler/pod"
+	"sigs.k8s.io/descheduler/pkg/framework"
+)
+
+const PluginName = "RemoveDuplicates"
+
+// RemoveDuplicatePods removes the duplicate pods on node. This strategy evicts all duplicate pods on node.
+// A pod is said to be a duplicate of other if both of them are from same creator, kind and are within the same
+// namespace, and have at least one container with the same image.
+// As of now, this strategy won't evict daemonsets, mirror pods, critical pods and pods with local storages.
+
+type RemoveDuplicates struct {
+	handle    framework.Handle
+	args      *componentconfig.RemoveDuplicatesArgs
+	podFilter podutil.FilterFunc
+}
+
+var _ framework.Plugin = &RemoveDuplicates{}
+var _ framework.DeschedulePlugin = &RemoveDuplicates{}
```

Comment:
RemoveDuplicates is one of the plugins that are aiming for balancing pods among nodes. Making [BalancePlugin](https://github.com/kubernetes-sigs/descheduler/blob/master/pkg/framework/types.go#L69) interface more suitable.

s/DeschedulePlugin/BalancePlugin.

Reply:
done